src/System.IO.Pipelines.Extensions/UnownedBufferReader.cs

```diff
@@ -66,6 +62,11 @@ public UnownedBufferReader()
 
         bool IReadableBufferAwaiter.IsCompleted => IsCompleted;
 
+        bool IReadableBufferAwaiter.IsCompletedSuccessfully => 
+            IsCompleted && 
+            _cancelledState == CancelledState.NotCancelled && 
+            !Reading.IsCanceled && !Reading.IsFaulted;
```

Comment:
Less allocaty, but still the slow path, I'll back out the changes for `IReadableBufferAwaiter` and raise an issue instead 😄